### PR TITLE
Refactor clickhouse admin integration tests

### PR DESCRIPTION
**Pull request by karencfv**
## Overview

This commit changes the way the clickhouse clusters are spun up for clickhouse-admin integration tests. Instead of multiple clusters being spun up for every test, now we start up a single cluster via nextest's "setup scripts" functionality.
This way, we no longer need to manually hard code so many ports per test.

## Caveats

Sadly, we still have to hardcode some ports, but at least this way it's only a set of ports, and we won't need to add more every time we want to test something new.

In order to get the teardown function working, I had to limit the tests to run on a single thread. They're a bit slower than before, but nothing too terrible. Setting up the cluster, running the tests and tearing down takes ~7s. The time to run the tests themselves is ~3s

### CODE REVIEW

File: clickhouse-admin/tests/integration_test.rs [hunk 1]
```diff
@@ -2,130 +2,89 @@
...
+            get_keeper_server_port(KeeperId(1)),
+            0,
+            0,
+        ),
     )
-    .with_log(log.clone());
+    .with_log(log);
 
-    let lgif = clickhouse_cli.lgif().await.unwrap();
+    let lgif = clickhouse_cli.lgif().await?;
```
Comment (andrewjstone) [inline clickhouse-admin/tests/integration_test.rs:58]: It's totally fine to use `?` here, but using unwrap will give you the proper line number for the failure, which may help debugging later.

File: dev-tools/clickhouse-cluster-dev/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
```
Comment (andrewjstone) [inline dev-tools/clickhouse-cluster-dev/src/main.rs:4]: Maybe add a doc comment here about how this should only be used for testing that doesn't write data to clickhouse, as that could make tests flaky.

**Review by andrewjstone — APPROVED** [review]
LGTM. All seems safe since we are just testing query functions for upgrade safety wrt parsing.

File: clickhouse-admin/tests/integration_test.rs (hunk 1 above)
Comment (karencfv) [reply clickhouse-admin/tests/integration_test.rs]: Huh, I wonder why I changed this. I didn't change the `.unwrap()` for a `?` anywhere else 😄

**Review by karencfv — COMMENTED** [review]
Thanks for the review @andrewjstone 🙇‍♀️

**Pull request merged**